squall/responses.py

```diff
@@ -74,16 +77,35 @@ def __init__(
         self.raw_headers = raw_headers = init_headers(
             body, self.charset, self.media_type, headers
         )
-        self.send_start = {
+        self.send_start: Dict[str, Any] = {
             "type": "http.response.start",
             "status": status_code,
             "headers": raw_headers,
         }
         self.send_body = {"type": "http.response.body", "body": body}
 
     async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
+        compression: Optional[Compression] = scope["app"].compression
+        send_body: Dict[str, Any] = self.send_body
+
+        if (
+            scope["type"] == "http"
+            and compression
+            and len(send_body["body"]) > compression.minimum_size
+        ):
+            accept_encoding = self.request.headers.get("Accept-Encoding", "")
+            for backend in compression.backends:
+                if backend.encoding_name in accept_encoding:
+                    body = backend.compress(send_body["body"], compression.level)
+                    headers = MutableHeaders(raw=self.send_start["headers"])
+                    headers["Content-Encoding"] = backend.encoding_name
+                    headers["Content-Length"] = str(len(body))
+                    headers.add_vary_header("Accept-Encoding")
```

Comment:
Something strange here. 
Headers assign is not explicit

Reply:
can't change right now
because initially self.send_start["headers"] is list of tuples